src/Core/View.js

```diff
@@ -217,11 +234,14 @@ function _preprocessLayer(view, layer, provider) {
  * });
  *
  * // Add Elevation Layer
- * view.addLayer({
+ * var layer = view.addLayer({
  *      type: 'elevation',
  *      id: 'iElevation',
  * });
  *
+ * // React to preprocessing
+ * layer.whenReady.then(() => { ... });
```

Comment:
Right.
I'll go with the more used pattern: if in all examples we always use `addLayer.whenReady` then I'd return the promise directly (and the user can still retrieve the `layer` instance using its `id`).
Otherwise let's keep `addLayer` as is.